pint/unit.py

```diff
@@ -1145,6 +1145,36 @@ def wrapper(*values, **kw):
             return wrapper
         return decorator
 
+    def check(self, *args):
+        """Decorator to for quantity type checking for function inputs.
+
+        Use it to ensure that the decorated function input parameters match
+        the expected type of pint quantity.
+
+        Use None to skip argument checking.
+
+        :param args: iterable of input units.
+        :return: the wrapped function.
+        :raises:
+            :class:`TypeError` if the parameters don't match dimensions
+        """
+        dimensions = [self.get_dimensionality(dim) for dim in args]
+
+        def decorator(func):
+            assigned = tuple(attr for attr in functools.WRAPPER_ASSIGNMENTS if hasattr(func, attr))
+            updated = tuple(attr for attr in functools.WRAPPER_UPDATES if hasattr(func, attr))
+
+            @functools.wraps(func, assigned=assigned, updated=updated)
+            def wrapper(*values, **kwargs):
+                for dim, value in itertools.izip_longest(dimensions, values):
```

Comment:
Two things about this.
1) izip_longest does not exist in Python 3+ (it is called zip_longest). Please add it via the compat module
2) In `wraps` we used zip, but actually I think that zip_longest is better. Would you mind submitting another PR fixing that.